### PR TITLE
Fix for Issue #384: typehints_defaults = "braces-after" fails for a multiline `:param:` entry

### PR DESCRIPTION
**Pull request by tikuma-lsuhsc**
Here is another quick fix.

The fix advances the line to the last line of the last `:param:` paragraph by detecting either that the next line is empty or has no indentation and placing the default at the end of the last nonempty line.

I've read somewhere else that rst docstring spec does not explicitly mention anything about multiline argument descriptions, but I understand it is widely accepted to use hanging indentation.

@gaborbernat & @hoodmane - One thing I'm not sure of is the handling of a multi-paragraph `:param:` description. It is likely rare to document in such a way. But when it does, should the default be placed at the end of the first or the last paragraph? 

I've gone with the latter for now, but I can see myself preferring the default to appear on the first paragraph when it happens. (2nd+ paragraphs for in-depth descriptions.) Ultimately, we can add a new option for it, but a default behavior must be set regardless. The current on-the-last-paragraph implementation is simply the last one I thought of, and I can go either way. Let me know what you guys think, and I'd be happy to modify to suit your preference.

### CODE REVIEW

**Comment by gaborbernat** [comment]
CI broken 😎

**Comment by tikuma-lsuhsc** [comment]
@gaborbernat - Working on it... too many nesting? y'all are picky!/s

File: src/sphinx_autodoc_typehints/__init__.py [hunk 1]
```diff
@@ -749,14 +749,33 @@ def _inject_signature(  # noqa: C901
...
 
+def _append_default(
+    app: Sphinx, lines: list[str], insert_index: int, type_annotation: str, formatted_default: str
+) -> str:
+    if app.config.typehints_defaults.endswith("after"):
+        # advance the index to the end of the :param: paragraphs
+        # (terminated by a line with no indentation)
+        # append default to the last nonempty line
+        nlines = len(lines)
+        i = insert_index + 1
```
Comment (gaborbernat) [inline src/sphinx_autodoc_typehints/__init__.py:765]: The only place I'm okay using single letter variable is within list compressions. Anywhere else single letter variables are so abstract that they are really hard to read. Please use more descriptive names.

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: src/sphinx_autodoc_typehints/__init__.py (hunk 1 above)
Comment (tikuma-lsuhsc) [reply src/sphinx_autodoc_typehints/__init__.py]: done

**Review by gaborbernat — APPROVED** [review]

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**